modules/merkleIdSystem.js

```diff
@@ -8,8 +8,40 @@
 import * as utils from '../src/utils.js'
 import {ajax} from '../src/ajax.js';
 import {submodule} from '../src/hook.js'
+import { getStorageManager } from '../src/storageManager.js';
 
 const MODULE_NAME = 'merkleId';
+const SESSION_COOKIE_NAME = '_svsid';
+
+export const storage = getStorageManager();
+
+function getSession(configParams) {
+  let session = null;
+  if (typeof configParams.sv_session !== 'string') {
+    session = configParams.sv_session;
+  } else {
+    session = readCookie() || readFromLocalStorage();
+  }
+  return session;
+}
+
+function readCookie() {
+  return storage.cookiesAreEnabled() ? storage.getCookie(SESSION_COOKIE_NAME) : null;
+}
+
+function readFromLocalStorage() {
+  return storage.localStorageIsEnabled() ? storage.getDataFromLocalStorage(SESSION_COOKIE_NAME) : null;
+}
+
+function constructUrl(configParams) {
+  const session = getSession(configParams);
+  let url = `${configParams.endpoint}?vendor=${configParams.vendor}&sv_cid=${configParams.sv_cid}&sv_domain=${configParams.sv_domain}&sv_pubid=${configParams.sv_pubid}`;
+  if (session) {
+    url.append(`?sv_session=${session}`);
```

Comment:
i think this start with an `&` and not a `?` since it's not the first querystring param in the list